backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java

```diff
@@ -2012,6 +2030,65 @@ public void testEndpointSubtypeIsOnlyAllowedWhenRequired() {
                 .statusCode(400);
     }
 
+    /**
+     * Tests that when sending a payload to the "/test" REST endpoint, a Kafka message is sent with a test event for
+     * that endpoint.
+     */
+    @Test
+    void testEndpointTest() {
+        final String accountId = "test-endpoint-test-account-number";
+        final String orgId = "test-endpoint-test-org-id";
+
+        final Endpoint createdEndpoint = this.resourceHelpers.createEndpoint(accountId, orgId, EndpointType.CAMEL);
+
+        final String identityHeaderValue = TestHelpers.encodeRHIdentityInfo(accountId, orgId, "user-name");
+        final Header identityHeader = TestHelpers.createRHIdentityHeader(identityHeaderValue);
+
+        MockServerConfig.addMockRbacAccess(identityHeaderValue, MockServerConfig.RbacAccess.FULL_ACCESS);
+
+        // Call the endpoint under test.
+        final String path = String.format("/endpoints/%s/test", createdEndpoint.getId());
+        given()
+            .header(identityHeader)
+            .when()
+            .post(path)
+            .then()
+            .statusCode(204);
+
+        // Capture the sent payload to verify it.
+        final ArgumentCaptor<EndpointTestRequest> capturedPayload = ArgumentCaptor.forClass(EndpointTestRequest.class);
+        Mockito.verify(this.endpointTestService).testEndpoint(capturedPayload.capture());
+
+        final EndpointTestRequest sentPayload = capturedPayload.getValue();
+
+        Assertions.assertEquals(createdEndpoint.getId(), sentPayload.endpointUuid, "the sent endpoint UUID in the payload doesn't match the one from the fixture");
+        Assertions.assertEquals(orgId, sentPayload.orgId, "the sent org id in the payload doesn't match the one from the fixture");
+    }
+
+    /**
+     * Tests that when the "test endpoint" handler is called with an endpoint
+     * UUID that doesn't exist, a not found response is returned.
+     */
+    @Test
+    void testEndpointTestNotFound() {
+        final String accountId = "test-endpoint-test-account-number";
+        final String orgId = "test-endpoint-test-org-id";
+
+        final String identityHeaderValue = TestHelpers.encodeRHIdentityInfo(accountId, orgId, "user-name");
+        final Header identityHeader = TestHelpers.createRHIdentityHeader(identityHeaderValue);
+
+        MockServerConfig.addMockRbacAccess(identityHeaderValue, MockServerConfig.RbacAccess.FULL_ACCESS);
+
+        // Call the endpoint under test.
+        final String path = String.format("/endpoints/%s/test", UUID.randomUUID());
+        given()
+                .header(identityHeader)
+                .when()
+                .post(path)
+                .then()
+                .statusCode(404);
```

Comment:
If you apply my other comment which adds an error message, please check that message in this test.

Reply:
Done!